src/main/java/io/cdap/plugin/sap/odata/ODataUtil.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.sap.odata;
+
+import org.apache.olingo.commons.api.edm.annotation.EdmExpression;
+import org.apache.olingo.commons.api.edm.provider.annotation.CsdlConstantExpression;
+import org.apache.olingo.commons.api.edm.provider.annotation.CsdlDynamicExpression;
+import org.apache.olingo.commons.api.edm.provider.annotation.CsdlExpression;
+
+/**
+ * OData utility methods.
+ */
+public class ODataUtil {
+
+  private ODataUtil() {
+    throw new AssertionError("Should not instantiate static utility class.");
+  }
+
+  public static EdmExpression.EdmExpressionType typeOf(CsdlExpression expression) {
+    return expression.isConstant() ? typeOf(expression.asConstant()) : typeOf(expression.asDynamic());
+  }
+
+  public static EdmExpression.EdmExpressionType typeOf(CsdlConstantExpression constant) {
+    // type names match
+    return EdmExpression.EdmExpressionType.valueOf(constant.getType().name());
+  }
+
+  public static EdmExpression.EdmExpressionType typeOf(CsdlDynamicExpression dynamic) {
+    if (dynamic.isPath()) {
```

Comment:
it seems odd that you would need to do this mapping, there isn't a way to get the type directly from a CsdlDynamicExpression? How are all these isX() methods implemented underneath if they aren't checking some enum?

Reply:
Currently, we use Csdl model classes that are not checking enum.
For example, the method above [implemented as](https://github.com/apache/olingo-odata4/blob/master/lib/commons-api/src/main/java/org/apache/olingo/commons/api/edm/provider/annotation/CsdlDynamicExpression.java#L205):
```
  /**
   * Returns true if the expression is a edm:Path expression
   * @return true if the expression is a edm:Path expression
   */
  public boolean isPath() {
    return this instanceof CsdlPath;
  }
```
This mapping is added to use `switch-case` instead of duplication of such checks.

Reply:
i see, that's unfortunate